build-plugins/build-support/src/main/kotlin/aws/sdk/kotlin/gradle/plugins/artifactsizemetrics/CollectDelegatedArtifactSizeMetrics.kt

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package aws.sdk.kotlin.gradle.plugins.artifactsizemetrics
+
+import aws.sdk.kotlin.gradle.util.AwsSdkGradleException
+import aws.sdk.kotlin.services.s3.S3Client
+import aws.sdk.kotlin.services.s3.listObjects
+import aws.sdk.kotlin.services.s3.model.GetObjectRequest
+import aws.smithy.kotlin.runtime.content.decodeToString
+import kotlinx.coroutines.runBlocking
+import org.gradle.api.DefaultTask
+import org.gradle.api.file.RegularFileProperty
+import org.gradle.api.tasks.OutputFile
+import org.gradle.api.tasks.TaskAction
+
+/**
+ * Collects the artifact size metrics for a projects subprojects from S3, combines them and outputs them to a file.
+ * This task should typically be run after codebuild gathers metrics and puts them in S3 during CI but can also be used to
+ * query the metrics bucket if you modify the file key filter.
+ */
+internal abstract class CollectDelegatedArtifactSizeMetrics : DefaultTask() {
+    /**
+     * The file where the artifact size metrics will be stored, defaults to /build/reports/metrics/artifact-size-metrics.csv
+     */
+    @get:OutputFile
+    abstract val metricsFile: RegularFileProperty
+
+    init {
+        metricsFile.convention(project.layout.buildDirectory.file(OUTPUT_PATH + "artifact-size-metrics.csv"))
+    }
+
+    @TaskAction
+    fun collect() {
+        val pullRequestNumber = if (project.hasProperty("pullRequest")) {
+            project.property("pullRequest")
+                .toString()
+                .let { releaseProperty ->
+                    releaseProperty.ifEmpty { // "-PpullRequest=" (no value set)
+                        null
+                    }
+                }
+        } else {
+            null
+        }
+
+        val releaseTag = if (project.hasProperty("release")) {
+            project.property("release")
+                .toString()
+                .let { releaseProperty ->
+                    releaseProperty.ifEmpty { // "-Prelease=" (no value set)
+                        null
+                    }
+                }
+        } else {
+            null
+        }
+
+        val identifier = pullRequestNumber ?: releaseTag ?: throw AwsSdkGradleException("Please specify a pull request or release number")
+
+        val artifactSizeMetricsFileKeys = getFileKeys() ?: throw AwsSdkGradleException("Unable to list objects from artifact size metrics bucket")
+
+        val pluginConfig = this.project.rootProject.extensions.getByType(ArtifactSizeMetricsPluginConfig::class.java)
+
+        val relevantArtifactSizeMetricsFileKeys = artifactSizeMetricsFileKeys.filter {
+            it?.startsWith("[TEMP]${pluginConfig.projectRepositoryName}-$identifier-") == true
+        }
```

Comment:
suggestion: You can configure `prefix` in the S3 `ListObjects` request instead of filtering it yourself